main.go

```diff
@@ -1,5 +1,59 @@
 package main
 
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+	"go.uber.org/zap"
+)
 
 func main() {
+	//logger := logr.Logger{}
+	ctx := context.Background()
+	configFilePath := os.Getenv("CONFIG_FILE_PATH")
+	if configFilePath == "" {
+		log.Fatal("CONFIG_FILE_PATH environment variable not set")
+	}
+	// parse configuration from yaml file
+	conf, err := config.LoadAndValidateConfig(configFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	zapConfig := zap.Config{
+		Encoding:         "json",
+		EncoderConfig:    zap.NewProductionEncoderConfig(),
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+	logger, err := zapConfig.Build()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dbInstance, err := db.NewClient(ctx, &conf.DB)
+	if err != nil {
+		logger.Fatal(err.Error())
```

Comment:
Because the defer is not set yet you need to close the client in case of an error 
```suggestion
		dbInstance.close()
		logger.Fatal(err.Error())
```

Reply:
- I don't think so, there can be scenarios where `dbInstance` can be `nil` so we need to defer after ensuring no error, (unless I understood the ask differently)

Reply:
So do this:
```suggestion
		if dbInstance != nil {
			dbInstance.close()
		}
		logger.Fatal(err.Error())
```

Reply:
This is needed unless we have guarantees
 that `db.NewClient` either returns an instance or error (never both). This is not the case with the current revision of the code

Reply:
This is not required incase of error dbClient will always be nil, I don't think anyone will return both

Reply:
👍